Firestore/core/test/unit/model/target_index_matcher_test.cc

```diff
@@ -139,10 +139,13 @@ void ValidateDoesNotServeTarget(const core::Query& query,
 void ValidateBuildTargetIndexCreateFullMatchIndex(const core::Query& query) {
   const core::Target& target = query.ToTarget();
   TargetIndexMatcher matcher(target);
-  FieldIndex expected_index = matcher.BuildTargetIndex();
-  EXPECT_TRUE(matcher.ServedByIndex(expected_index));
+  EXPECT_FALSE(matcher.HasMultipleInequality());
+  absl::optional<FieldIndex> actual_index = matcher.BuildTargetIndex();
+  EXPECT_TRUE(actual_index.has_value());
```

Comment:
nit: Change this `EXPECT_TRUE` to `ASSERT_TRUE` because if `has_value()` unexpectedly returns false you do _not_ want to call `value()` on the next line because doing so is undefined behavior.